Internet access required when running the bedrock-access-gateway

**Is your feature request related to a problem? Please describe.**
I use bedrock access gateway in a close environment, and found that It try to download the cl100k_base.tiktoken file.  which will fail and cause the lambda fail to process the request

**Describe the feature you'd like**
is it possible to embded the token file in the lambda image so it doesnt request external access?

error log from lambda

[ERROR] ConnectTimeout: HTTPSConnectionPool(host='openaipublic.blob.core.windows.net', port=443): Max retries exceeded with url: /encodings/cl100k_base.tiktoken (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0xffff8df0d250>, 'Connection to openaipublic.blob.core.windows.net timed out. (connect timeout=None)'))
Traceback (most recent call last):
File "/var/lang/lib/python3.12/importlib/init.py", line 90, in import_module
return _bootstrap._gcd_import(name[level:], package, level)
File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
File "<frozen importlib._bootstrap_external>", line 999, in exec_module
File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
File "/var/task/api/app.py", line 10, in <module>
from api.routers import model, chat, embeddings
File "/var/task/api/routers/model.py", line 6, in <module>
from api.models.bedrock import BedrockModel
File "/var/task/api/models/bedrock.py", line 76, in <module>
ENCODER = tiktoken.get_encoding("cl100k_base")
File "/var/lang/lib/python3.12/site-packages/tiktoken/registry.py", line 73, in get_encoding
enc = Encoding(**constructor())
File "/var/lang/lib/python3.12/site-packages/tiktoken_ext/openai_public.py", line 72, in cl100k_base
mergeable_ranks = load_tiktoken_bpe(
File "/var/lang/lib/python3.12/site-packages/tiktoken/load.py", line 147, in load_tiktoken_bpe
contents = read_file_cached(tiktoken_bpe_file, expected_hash)
File "/var/lang/lib/python3.12/site-packages/tiktoken/load.py", line 64, in read_file_cached
contents = read_file(blobpath)
File "/var/lang/lib/python3.12/site-packages/tiktoken/load.py", line 25, in read_file
resp = requests.get(blobpath)
File "/var/lang/lib/python3.12/site-packages/requests/api.py", line 73, in get
return request("get", url, params=params, **kwargs)
File "/var/lang/lib/python3.12/site-packages/requests/api.py", line 59, in request
return session.request(method=method, url=url, **kwargs)
File "/var/lang/lib/python3.12/site-packages/requests/sessions.py", line 589, in request
resp = self.send(prep, **send_kwargs)
File "/var/lang/lib/python3.12/site-packages/requests/sessions.py", line 703, in send
r = adapter.send(request, **kwargs)
File "/var/lang/lib/python3.12/site-packages/requests/adapters.py", line 688, in send
raise ConnectTimeout(e, request=request)


reference link talk about using this in offline environment

https://stackoverflow.com/questions/76106366/how-to-use-tiktoken-in-offline-mode-computer

--- a/src/Dockerfile
+++ b/src/Dockerfile
@@ -9,6 +9,10 @@ COPY requirements.txt .
 
 RUN pip3 install -r requirements.txt -U --no-cache-dir
 
+# Preload tiktoken encoding: https://github.com/aws-samples/bedrock-access-gateway/issues/118
+ENV TIKTOKEN_CACHE_DIR=/var/task/.cache/tiktoken
+RUN python3 -c 'import tiktoken_ext.openai_public as tke; tke.cl100k_base()'
+
 # Lambda Web Adapter requires overriding the Lambda base image entrypoint
 # to run the web app directly instead of the Lambda runtime handler
 ENTRYPOINT []

